spec/policy/logging/logging_spec.lua

```diff
@@ -36,4 +40,187 @@ describe('Logging policy', function()
+        local logging = LoggingPolicy.new({
+          custom_logging = "foo",
+          condition = {
+            operations={{op="==", match="{{ foo }}", match_type="liquid", value="fooValue", value_type="plain"}},
+            combine_op="and"
+          }})
+        logging:log(ctx)
+        assert.equals(0, ngx.var.access_logs_enabled)
+        assert.equals(1, ngx.var.extended_access_logs_enabled)
+        assert.equals("foo", ngx.var.extended_access_log)
+      end)
+
+      it("Validate default combine_op", function()
+        local logging = LoggingPolicy.new({
+          custom_logging = "foo",
+          condition = {
+            operations={{op="==", match="{{ foo }}", match_type="liquid", value="fooValue", value_type="plain"}}
+          }})
+        logging:log(ctx)
+        assert.equals(0, ngx.var.access_logs_enabled)
+        assert.equals(1, ngx.var.extended_access_logs_enabled)
+        assert.equals("foo", ngx.var.extended_access_log)
+      end)
+
+      it("Or combination match one", function()
+        local logging = LoggingPolicy.new({
+          custom_logging = "foo",
+          condition = {
+            operations={
+              {op="==", match="{{ invalid }}", match_type="liquid", value="fooValue", value_type="plain"},
+              {op="==", match="{{ foo }}", match_type="liquid", value="fooValue", value_type="plain"}
+            },
+            combine_op="or"
+          }})
+        logging:log(ctx)
+        assert.equals(0, ngx.var.access_logs_enabled)
+        assert.equals(1, ngx.var.extended_access_logs_enabled)
+        assert.equals("foo", ngx.var.extended_access_log)
+      end)
+
+      it("No Match combination", function()
+        local logging = LoggingPolicy.new({
+          custom_logging = "foo",
+          condition = {
+            operations={{op="==", match="{{ invalid }}", match_type="liquid", value="fooValue", value_type="plain"}},
+            combine_op="and"
+          }})
+        logging:log(ctx)
+        assert.equals(0, ngx.var.access_logs_enabled)
+        assert.equals(0, ngx.var.extended_access_logs_enabled)
+        assert.is_nil(ngx.var.extended_access_log)
+      end)
+
+      describe("JSON Logs", function()
+
+        local expected_json = cjson.encode({foo="fooValue", bar="barValue"})
+
+        it("valid config", function()
+          local logging = LoggingPolicy.new({
+            enable_json_logs = true,
+            json_object_config = {
+              { key = "foo", value="{{foo}}", value_type="liquid"},
+              { key = "bar", value="{{req.headers.bar}}", value_type="liquid"},
+            }
+          })
+          spy.on(logging, "log_dump_json")
+          logging:log(ctx)
+
+          assert.equals(0, ngx.var.access_logs_enabled)
+          assert.equals(1, ngx.var.extended_access_logs_enabled)
+          assert.equals(ngx.var.extended_access_log, expected_json)
+          assert.spy(logging.log_dump_json).was.called()
```

Comment:
Why is this needed? The previous assert already verifies that a JSON was generated, no?

Reply:
just to make sure that the call to the correct function for future changes.